### PR TITLE
boards: arm: stm32f746g_disco: Add support for FMC SDRAM and LTDC display

### PR DESCRIPTION
**Pull request by tmilkovic51**
- This board uses the same LCD panel like STM32H7B3I_DK board implemented in PR #43965, so same LCD parameters are used in board device tree
- FMC SDRAM device tree parameters were taken from STM32CubeIDE example project for this board
- On STM32F7 series, LTDC pixel clock is supplied by PLLSAI:
  - There has been a temporary fix for configuring PLLSAI on STM32F4 and STM32F7 series in LTDC driver on commit https://github.com/zephyrproject-rtos/zephyr/commit/b8604024f7ceadafd5778e78640102023b6ebc33 
  - Since STM32F746G_DISCO board is the only F7 board with RGB display interface supported by Zephyr, this PR changes the hardcoded F7 series PLLSAI configuration in LTDC driver, so that 9.6 MHz pixel clock is output if 25 MHz HSE is used
  - Once PR #42097 is merged, and proper configuration for PLLSAI is implemented in `clock_control` driver, the PLLSAI hack can be removed from LTDC driver
- Tested with `samples/drivers/display` and `samples/subsys/display/lvgl` samples

### CODE REVIEW

File: boards/arm/stm32f746g_disco/Kconfig.defconfig [hunk 1]
```diff
@@ -15,6 +15,9 @@ config NET_L2_ETHERNET
 
 endif # NETWORKING
 
+config KSCAN
+	default y if DISPLAY
+
```
Comment (erwango) [inline boards/arm/stm32f746g_disco/Kconfig.defconfig:20]: Let's group symbols that depends on DISPLAY
```
if DISPLAY

config KSCAN
	default y

# MEMC needs to be enabled in order to store
# display buffer to external SDRAM connected to FMC
config MEMC
	default y

endif
```

File: boards/arm/stm32f746g_disco/Kconfig.defconfig [hunk 2]
```diff
@@ -15,6 +15,9 @@ config NET_L2_ETHERNET
...
 config I2C
 	default y if KSCAN
```
Comment (erwango) [inline boards/arm/stm32f746g_disco/Kconfig.defconfig:22]: This one could be put under `if KSCAN` while you're at.

File: boards/arm/stm32f746g_disco/stm32f746g_disco.dts [hunk 3]
```diff
@@ -7,6 +7,7 @@
 /dts-v1/;
 #include <st/f7/stm32f746Xg.dtsi>
 #include <st/f7/stm32f746nghx-pinctrl.dtsi>
+#include <dt-bindings/display/stm32_ltdc.h>
```
Comment (erwango) [inline boards/arm/stm32f746g_disco/stm32f746g_disco.dts:10]: Move this to the right *.dtsi file(s)

**Review by erwango — CHANGES_REQUESTED** [review]
Thanks for this contribution. A couple of changes required

File: boards/arm/stm32f746g_disco/Kconfig.defconfig (hunk 2 above)
Comment (tmilkovic51) [reply boards/arm/stm32f746g_disco/Kconfig.defconfig]: Grouped

File: boards/arm/stm32f746g_disco/stm32f746g_disco.dts (hunk 3 above)
Comment (erwango) [reply boards/arm/stm32f746g_disco/stm32f746g_disco.dts]: Thanks

> If that is too much for this PR, I can move the commits doing that to another branch and open a different PR

That should be fine, txs

**Review by erwango — APPROVED** [review]

**Review by FRASTM — APPROVED** [review]

**Pull request merged**